### DISCUSSION TITLE
Proxy self hosted

### DISCUSSION THREAD

**Discussion by robflate** [hoppscotch/hoppscotch#1380] [Q&A] [answered]
In layman's terms can someone explain the purpose of the proxy? I currently use Postman to test APIs but understand that browsers are treated differently and can have problems with CORS and have other limitations that an app doesn't. Is the proxy to overcome this?

I've installed Hoppscotch in Docker with Traefik as the reverse proxy and oauth with Google via Traefik's forweard auth. The only issue is I get Network errors for unsuccessful connections. 

For example, Traefik allows access to `/api` in a certain app if the query contains a known key, otherwise it redirect to a Google oauth login page. In Postman, a failed request returns the source code for the Google oauth login page but in Hoppscotch it shows a network error.

I understand the built in proxy won't work for self hosted setups. I tried the proxyscotch image but it just forwards to a page saying Postwoman is now Hoppscotch.

I do this for a hobby so don't really understand what I'm sure is obvious to most people with a good understanding of this stuff. Thanks.

**Comment by liyasthomas** [comment, marked as answer]
[Proxyscotch](https://github.com/hoppscotch/proxyscotch) - is the official distribution of default proxy server used in Hoppscotch project. In simple terms, proxy mode acts as a middleware between our request `origin` (web app) and request `endpoint` (URL).

By default Hoppscotch generate request from client side (browser). But there're endpoints which restrict requests generated from client origins. This restrictions can be solved with proxy mode since when turned on, all requests are generated from server (proxy URL). This fixes `CORS` issues.

> I tried the proxyscotch image but it just forwards to a page saying Postwoman is now Hoppscotch.

Thanks for noticing this, we'll fix this ASAP.
